### ISSUE TITLE
Vue SFC: Incorrect minified JS

### ISSUE THREAD

**Issue by SteffenL** [parcel-bundler/parcel#1350]
# 🐛 bug report

I apologize if this is not the right place to file the report.

When building/minifying my Vue SFC file, the minified code seems to have a bug causing a `TypeError` to be thrown in the browser. It only takes slight modifications of the original script to "fix" the error. It may be possible to further narrow down the scope of the bug, but it has been quite time-consuming for me already to strip my app down to this.

Could this be related to #1344?

## 🎛 Configuration (.babelrc, package.json, cli command)

.babelrc: N/A

package.json:
```js
{
  "name": "testapp",
  "version": "1.0.0",
  "description": "testapp",
  "private": true,
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "parcel build src/index.html",
    "dev": "parcel src/index.html"
  },
  "author": "Steffen",
  "license": "ISC",
  "dependencies": {
    "vue": "^2.5.16"
  },
  "devDependencies": {
    "@vue/component-compiler-utils": "^1.2.1",
    "parcel-bundler": "^1.8.1",
    "vue-template-compiler": "^2.5.16"
  }
}
```

cli command:
```
parcel build src/index.html
```

## 🤔 Expected Behavior

Generate correct minified code.

## 😯 Current Behavior

Generates incorrect minified code.

Full error message and stack trace:
```
TypeError: e.default is undefined
Stack trace:
mounted@http://localhost:8000/src.4b8e3a3d.js:7:226
bn@http://localhost:8000/src.4b8e3a3d.js:3:20842
insert@http://localhost:8000/src.4b8e3a3d.js:3:29531
x@http://localhost:8000/src.4b8e3a3d.js:3:45587
Vo/<@http://localhost:8000/src.4b8e3a3d.js:3:46960
vn/t.prototype._update@http://localhost:8000/src.4b8e3a3d.js:3:18781
hn/<@http://localhost:8000/src.4b8e3a3d.js:3:19671
parcelRequire<[4]</Pn.prototype.get@http://localhost:8000/src.4b8e3a3d.js:3:22172
Pn@http://localhost:8000/src.4b8e3a3d.js:3:22097
hn@http://localhost:8000/src.4b8e3a3d.js:3:19651
parcelRequire<[4]</Pr.prototype.$mount@http://localhost:8000/src.4b8e3a3d.js:3:63217
Ir/t.prototype._init@http://localhost:8000/src.4b8e3a3d.js:3:32681
Pr@http://localhost:8000/src.4b8e3a3d.js:3:33641
parcelRequire<[2]<@http://localhost:8000/src.4b8e3a3d.js:9:125
i@http://localhost:8000/src.4b8e3a3d.js:1:395
parcelRequire<@http://localhost:8000/src.4b8e3a3d.js:1:696
@http://localhost:8000/src.4b8e3a3d.js:1:15
```

Minified (prettified) code with some comments:
```js
  function (require, module, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", {
      value: !0
    });
    var t = require("./Auth"),
      e = u(t);
      // e = Object { __esModule: true, default: Object { isAuthenticated: isAuthenticated() } }

    function u(t) {
      return t && t.__esModule ? t : {
        default: t
      }
    }
    exports.default = {
      data: function () {
        return {}
      },
      mounted: function () {
        // e = Object { data: data(), mounted: mounted() ... }
        var t = this;
        return e.default.isAuthenticated().then(function () { // TypeError: e.default is undefined
          t.setStage()
        })
      },
      methods: {
        setStage: function () {}
      }
    };
    var e = exports.default || module.exports;
    // e = Object { data: data(), mounted: mounted() ... }
    "function" == typeof e && (e = e.options), Object.assign(e, {
      render: function () {
        var e = this.$createElement;
        return (this._self._c || e)("div", {
          attrs: {
            id: "app"
          }
        })
      },
      staticRenderFns: [],
      _compiled: !0,
      _scopeId: null,
      functional: void 0
    });
  }, {
    "./Auth": 5
  }
```

## 🔦 Context

This is a blocker for production-ready apps and minified code in general.

## 💻 Code Sample

https://github.com/SteffenL/parcel-vue-bug-repro

Building the code:
```
npm install
npm run build
```

Testing with Python 2 HTTP server:
```
cd dist
python -m SimpleHTTPServer
```

Testing with Python 3 HTTP server:
```
cd dist
python -m http.server
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.8.1
| Node             | 8.11.1
| npm/Yarn         | 5.6.0
| Operating System | Windows 10

**Comment by SteffenL** [comment]
After looking more at #1344, I believe this issue might be a duplicate. Sorry about that if so.

Any ideas exactly which dependency has the bug so that a report can be made in the right place if necessary?

**Comment by DeMoorJasper** [comment]
It's probably related or the same as #1344, I'll leave this open.

Like I've said in #1344 experiment a bit with uglifyrc config, let us know if you found something that could help resolve/prevent this :)

**Comment by SteffenL** [comment]
@DeMoorJasper Thank you! I started experimenting with this earlier, then realized that nothing I do changes the output. Do I need to do something special to make Parcel use the `.uglifyrc` file?

I added this to my test repository:
```
{
    compress: false
    mangle: false,
    output: {
        beautify: false,
        comments: true
    }
}
```

I also made a longer file with many of the default options changed:
https://gist.github.com/SteffenL/257c15709aacaf1fa90ee83f9c65eac0

**Comment by SteffenL** [comment]
Parcel did not do anything with the `.uglifyrc` file until I cleared the cache or used the `--no-cache` parameter. I can now start experimenting again.

**Comment by SteffenL** [comment]
The following config gets rid of the error both in the test app as well as my main app. The main app is however still has major broken parts without throwing any errors:
```
{
    mangle: false
}
```

**Comment by DeMoorJasper** [comment]
Alright seems to be resolved than.

The cache works by adding the config as dependencies to the asset, so it doesn't notice new files only updates.

I haven't been able to reproduce the freeze bug, feel free to append to the issue related to that issue if you can help me get a consistently failing project.

**Comment by SteffenL** [comment]
I am not sure the actual problem has been resolved, but yes, the error described here disappeared. The real problem probably still exists, just that name-mangling made it worse. The only thing that actually works for me so far is using `--no-minify`, which incrases my bundle size by 1-2 megabytes.

I will keep trying since I would like to keep using Parcel.

Regarding the freezing, I am trying to play with Parcel's code. So far, I have only learned that a promise returned by `Bundler#loadAsset` may be in "pending" state forever.

My main project freezes maybe half of the time I try to build, while the example project I provided here does not seem to freeze often during build.
